command/src/state.rs

```diff
@@ -401,4 +435,49 @@ mod tests {
 
    assert_eq!(diff, expected_diff);
   }
+
+  #[test]
+  fn application_ids_by_domain() {
+    let mut config = ConfigState::new();
+    let http_front_app1 = HttpFront {
+      app_id: String::from("MyApp_1"),
+      hostname: String::from("lolcatho.st"),
+      path_begin: String::from("")
+    };
+
+    let https_front_app1 = HttpsFront {
+      app_id: String::from("MyApp_1"),
+      hostname: String::from("lolcatho.st"),
+      path_begin: String::from(""),
+      fingerprint: CertFingerprint(vec!(0x00))
+    };
+
+    let http_front_app2 = HttpFront {
+      app_id: String::from("MyApp_2"),
+      hostname: String::from("lolcatho.st"),
+      path_begin: String::from("/api")
+    };
+
+    let https_front_app2 = HttpsFront {
+      app_id: String::from("MyApp_2"),
+      hostname: String::from("lolcatho.st"),
+      path_begin: String::from("/api"),
+      fingerprint: CertFingerprint(vec!(0x00))
+    };
+
+    let add_http_front_order_app1 = Order::AddHttpFront(http_front_app1);
+    let add_http_front_order_app2 = Order::AddHttpFront(http_front_app2);
+    let add_https_front_order_app1 = Order::AddHttpsFront(https_front_app1);
+    let add_https_front_order_app2 = Order::AddHttpsFront(https_front_app2);
+    config.handle_order(&add_http_front_order_app1);
+    config.handle_order(&add_http_front_order_app2);
+    config.handle_order(&add_https_front_order_app1);
+    config.handle_order(&add_https_front_order_app2);
+
+    let empty: Vec<AppId> = Vec::new();
```

Comment:
use hashsets to store the list of application ids in the test, to avoid errors likes this one: https://travis-ci.org/sozu-proxy/sozu/jobs/323972206#L877